### DISCUSSION TITLE
Search function transaction history

### DISCUSSION THREAD

**Discussion by MiKeF05** [iotaledger/firefly#655] [Features / Enhancements]
Maybe it would be great to have a **search function** in the transaction history. Because when someone tries to find a specific receive address among many others this could be really hard. A search function with the receive address could be good in this case.

**Comment by maxwellmattryan** [comment]
From @Phyloiota (#594):

## Description

A page inside a wallet profile that shows me all transactions that have happened in this profile.
It should be able to search and sort all transactions based on different filter - / sort - options. Also should be able to export those filtered tx as a file (csv export already planned Charlie said).
Basically it should work similar to windows file explorer.
Suggested structure / sort options 

<img width="554" alt="TX archieve" src="https://user-images.githubusercontent.com/77154511/111927495-e5115c80-8aeb-11eb-9754-78e3572ae2d0.png">


## Motivation

there might be users that do a lot of tx over time. The current view "latest transactions" doesnt give a good option to keep an overview - no timestamps visible etc...
After a while you have no idea which tx does belong to which action / time / type

## Requirements

Write a list of what you want this feature to do.

1.assign following attributes to every message in a profile: Account name /date /time /type (value or data) /Action (sending or receiving) / message id /value /sent to address /received from address
2. archieve this data and make it graphical visible and sortable filterable in an explorer style page
3. make selected data exportable as file

## Open questions (optional)

Dont know if you have planned it, but think thats needed and a big enhancement for the user

## Are you planning to do it yourself in a pull request?

No

**Comment by maxwellmattryan** [comment]
Just wanted to add that as it's related to searching messages for metadata ~

**Comment by nicole-obrien** [comment]
Added an issue here for exporting transaction history, it isn't quite what this initial discussion is about but it may be of interest if you wish to add comments https://github.com/iotaledger/firefly/issues/1539
